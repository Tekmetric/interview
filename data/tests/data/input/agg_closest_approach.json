[
    {
        "id": "2001943",
        "neo_reference_id": "2001943",
        "name": "1943 Anteros (1973 EC)",
        "name_limited": "Anteros",
        "designation": "1943",
        "nasa_jpl_url": "https://ssd.jpl.nasa.gov/tools/sbdb_lookup.html#/?sstr=2001943",
        "absolute_magnitude_h": 15.69,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 1.9344387205,
                "estimated_diameter_max": 4.3255364773
            },
            "meters": {
                "estimated_diameter_min": 1934.4387204589,
                "estimated_diameter_max": 4325.5364772538
            },
            "miles": {
                "estimated_diameter_min": 1.2020041222,
                "estimated_diameter_max": 2.6877629264
            },
            "feet": {
                "estimated_diameter_min": 6346.5839316303,
                "estimated_diameter_max": 14191.3930960332
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [
            {
                "close_approach_date": "1937-05-28",
                "close_approach_date_full": "1937-May-28 20:45",
                "epoch_date_close_approach": -1028603700000,
                "relative_velocity": {
                    "kilometers_per_second": "7.2023034001",
                    "kilometers_per_hour": "25928.2922403221",
                    "miles_per_hour": "16110.8324756601"
                },
                "miss_distance": {
                    "astronomical": "0.1232067869",
                    "lunar": "47.9274401041",
                    "kilometers": "18431472.889783903",
                    "miles": "11452786.1809597414"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "1949-05-22",
                "close_approach_date_full": "1949-May-22 08:44",
                "epoch_date_close_approach": -650474160000,
                "relative_velocity": {
                    "kilometers_per_second": "5.737872245",
                    "kilometers_per_hour": "20656.3400821748",
                    "miles_per_hour": "12835.0464249491"
                },
                "miss_distance": {
                    "astronomical": "0.0655647844",
                    "lunar": "25.5047011316",
                    "kilometers": "9808352.093249228",
                    "miles": "6094627.3791182264"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "1961-05-19",
                "close_approach_date_full": "1961-May-19 05:02",
                "epoch_date_close_approach": -272055480000,
                "relative_velocity": {
                    "kilometers_per_second": "5.4200576246",
                    "kilometers_per_hour": "19512.2074486009",
                    "miles_per_hour": "12124.1269004932"
                },
                "miss_distance": {
                    "astronomical": "0.0654705949",
                    "lunar": "25.4680614161",
                    "kilometers": "9794261.544672863",
                    "miles": "6085871.9182289894"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "1973-05-12",
                "close_approach_date_full": "1973-May-12 13:39",
                "epoch_date_close_approach": 106061940000,
                "relative_velocity": {
                    "kilometers_per_second": "5.5534662514",
                    "kilometers_per_hour": "19992.4785051979",
                    "miles_per_hour": "12422.5486578548"
                },
                "miss_distance": {
                    "astronomical": "0.1088201212",
                    "lunar": "42.3310271468",
                    "kilometers": "16279258.344661844",
                    "miles": "10115462.0751636872"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "2026-05-30",
                "close_approach_date_full": "2026-May-30 08:25",
                "epoch_date_close_approach": 1780129500000,
                "relative_velocity": {
                    "kilometers_per_second": "7.3908131497",
                    "kilometers_per_hour": "26606.9273389196",
                    "miles_per_hour": "16532.5099345637"
                },
                "miss_distance": {
                    "astronomical": "0.1321633953",
                    "lunar": "51.4115607717",
                    "kilometers": "19771362.428848011",
                    "miles": "12285354.9338081118"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "2038-05-23",
                "close_approach_date_full": "2038-May-23 09:26",
                "epoch_date_close_approach": 2158219560000,
                "relative_velocity": {
                    "kilometers_per_second": "5.7770350659",
                    "kilometers_per_hour": "20797.3262370978",
                    "miles_per_hour": "12922.6497388234"
                },
                "miss_distance": {
                    "astronomical": "0.066409537",
                    "lunar": "25.833309893",
                    "kilometers": "9934725.28288619",
                    "miles": "6173152.037921822"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "2050-05-20",
                "close_approach_date_full": "2050-May-20 09:34",
                "epoch_date_close_approach": 2536652040000,
                "relative_velocity": {
                    "kilometers_per_second": "5.4511566191",
                    "kilometers_per_hour": "19624.1638289019",
                    "miles_per_hour": "12193.6922413529"
                },
                "miss_distance": {
                    "astronomical": "0.0636292082",
                    "lunar": "24.7517619898",
                    "kilometers": "9518794.016506534",
                    "miles": "5914704.3333728092"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "2062-05-14",
                "close_approach_date_full": "2062-May-14 21:01",
                "epoch_date_close_approach": 2914866060000,
                "relative_velocity": {
                    "kilometers_per_second": "5.4684295958",
                    "kilometers_per_hour": "19686.3465448092",
                    "miles_per_hour": "12232.330162801"
                },
                "miss_distance": {
                    "astronomical": "0.0960103695",
                    "lunar": "37.3480337355",
                    "kilometers": "14362946.775112965",
                    "miles": "8924721.281230317"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "2115-06-01",
                "close_approach_date_full": "2115-Jun-01 17:52",
                "epoch_date_close_approach": 4588854720000,
                "relative_velocity": {
                    "kilometers_per_second": "7.5421402139",
                    "kilometers_per_hour": "27151.7047700493",
                    "miles_per_hour": "16871.0134444787"
                },
                "miss_distance": {
                    "astronomical": "0.1395876312",
                    "lunar": "54.2995885368",
                    "kilometers": "20882012.305865544",
                    "miles": "12975480.7658267472"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "2127-05-24",
                "close_approach_date_full": "2127-May-24 21:51",
                "epoch_date_close_approach": 4966869060000,
                "relative_velocity": {
                    "kilometers_per_second": "5.7506206992",
                    "kilometers_per_hour": "20702.2345170508",
                    "miles_per_hour": "12863.5634419975"
                },
                "miss_distance": {
                    "astronomical": "0.0650662915",
                    "lunar": "25.3107873935",
                    "kilometers": "9733778.617199105",
                    "miles": "6048289.569813449"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "2139-05-20",
                "close_approach_date_full": "2139-May-20 15:23",
                "epoch_date_close_approach": 5345191380000,
                "relative_velocity": {
                    "kilometers_per_second": "5.4042130229",
                    "kilometers_per_hour": "19455.1668825231",
                    "miles_per_hour": "12088.6841109766"
                },
                "miss_distance": {
                    "astronomical": "0.0683525331",
                    "lunar": "26.5891353759",
                    "kilometers": "10225393.360864497",
                    "miles": "6353764.8064521786"
                },
                "orbiting_body": "Earth"
            },
            {
                "close_approach_date": "2151-05-10",
                "close_approach_date_full": "2151-May-10 11:55",
                "epoch_date_close_approach": 5723006100000,
                "relative_velocity": {
                    "kilometers_per_second": "6.2680617592",
                    "kilometers_per_hour": "22565.0223330112",
                    "miles_per_hour": "14021.0273490869"
                },
                "miss_distance": {
                    "astronomical": "0.1481644522",
                    "lunar": "57.6359719058",
                    "kilometers": "22165086.458836814",
                    "miles": "13772746.0747994732"
                },
                "orbiting_body": "Earth"
            }
        ],
        "orbital_data": {
            "orbit_id": "776",
            "orbit_determination_date": "2024-12-22 05:25:41",
            "first_observation_date": "1973-03-10",
            "last_observation_date": "2024-04-10",
            "data_arc_in_days": 18659,
            "observations_used": 5765,
            "orbit_uncertainty": "0",
            "minimum_orbit_intersection": ".0622285",
            "jupiter_tisserand_invariant": "4.640",
            "epoch_osculation": "2460800.5",
            "eccentricity": ".2561027627674788",
            "semi_major_axis": "1.43032986682301",
            "inclination": "8.708729946912538",
            "ascending_node_longitude": "246.2936957403874",
            "orbital_period": "624.8168568146249",
            "perihelion_distance": "1.064018436260797",
            "perihelion_argument": "338.4423868560169",
            "aphelion_distance": "1.796641297385223",
            "perihelion_time": "2460548.575588835703",
            "mean_anomaly": "145.150994295364",
            "mean_motion": ".5761688342329845",
            "equinox": "J2000",
            "orbit_class": {
                "orbit_class_type": "AMO",
                "orbit_class_description": "Near-Earth asteroid orbits similar to that of 1221 Amor",
                "orbit_class_range": "1.017 AU < q (perihelion) < 1.3 AU"
            }
        },
        "is_sentry_object": false
    },
    {
        "links": {
            "self": "http://api.nasa.gov/neo/rest/v1/neo/2001980?api_key=DEMO_KEY"
        },
        "id": "2001980",
        "neo_reference_id": "2001980",
        "name": "1980 Tezcatlipoca (1950 LA)",
        "name_limited": "Tezcatlipoca",
        "designation": "1980",
        "nasa_jpl_url": "https://ssd.jpl.nasa.gov/tools/sbdb_lookup.html#/?sstr=2001980",
        "absolute_magnitude_h": 13.82,
        "estimated_diameter": {
            "kilometers": {
                "estimated_diameter_min": 4.5767266723,
                "estimated_diameter_max": 10.2338719537
            },
            "meters": {
                "estimated_diameter_min": 4576.7266723128,
                "estimated_diameter_max": 10233.8719537278
            },
            "miles": {
                "estimated_diameter_min": 2.8438452291,
                "estimated_diameter_max": 6.3590312498
            },
            "feet": {
                "estimated_diameter_min": 15015.5079355907,
                "estimated_diameter_max": 33575.6964606684
            }
        },
        "is_potentially_hazardous_asteroid": false,
        "close_approach_data": [],
        "orbital_data": {
            "orbit_id": "886",
            "orbit_determination_date": "2025-02-19 05:50:13",
            "first_observation_date": "1950-06-19",
            "last_observation_date": "2025-02-18",
            "data_arc_in_days": 27273,
            "observations_used": 8019,
            "orbit_uncertainty": "0",
            "minimum_orbit_intersection": ".244969",
            "jupiter_tisserand_invariant": "3.996",
            "epoch_osculation": "2460800.5",
            "eccentricity": ".3647331673141841",
            "semi_major_axis": "1.709381149955462",
            "inclination": "26.87003874017241",
            "ascending_node_longitude": "246.5434218693389",
            "orbital_period": "816.3131500366128",
            "perihelion_distance": "1.085913148985044",
            "perihelion_argument": "115.4822640392797",
            "aphelion_distance": "2.33284915092588",
            "perihelion_time": "2460520.971967240634",
            "mean_anomaly": "123.2738830543871",
            "mean_motion": ".4410072286399569",
            "equinox": "J2000",
            "orbit_class": {
                "orbit_class_type": "AMO",
                "orbit_class_description": "Near-Earth asteroid orbits similar to that of 1221 Amor",
                "orbit_class_range": "1.017 AU < q (perihelion) < 1.3 AU"
            }
        },
        "is_sentry_object": false
    }
]
