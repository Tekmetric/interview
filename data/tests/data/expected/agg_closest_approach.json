[
    {
        "id": "2001943",
        "neo_reference_id": "2001943",
        "name": "1943 Anteros (1973 EC)",
        "name_limited": "Anteros",
        "designation": "1943",
        "nasa_jpl_url": "https://ssd.jpl.nasa.gov/tools/sbdb_lookup.html#/?sstr=2001943",
        "absolute_magnitude_h": 15.69,
        "estimated_diameter_min_meters": 1934.4387204589,
        "estimated_diameter_max_meters": 4325.5364772538,
        "is_potentially_hazardous_asteroid": false,
        "close_approach_date": "2050-05-20",
        "close_approach_miss_distance_kilometers": "9518794.0165065340",
        "close_approach_velocity_kilometers_per_second": "5.4511566191",
        "first_observation_date": "1973-03-10",
        "last_observation_date": "2024-04-10",
        "observations_used": 5765,
        "orbital_period": "624.8168568146249"
    },
    {
        "id": "2001980",
        "neo_reference_id": "2001980",
        "name": "1980 Tezcatlipoca (1950 LA)",
        "name_limited": "Tezcatlipoca",
        "designation": "1980",
        "nasa_jpl_url": "https://ssd.jpl.nasa.gov/tools/sbdb_lookup.html#/?sstr=2001980",
        "absolute_magnitude_h": 13.82,
        "estimated_diameter_min_meters": 4576.7266723128,
        "estimated_diameter_max_meters": 10233.8719537278,
        "is_potentially_hazardous_asteroid": false,
        "close_approach_date": null,
        "close_approach_miss_distance_kilometers": null,
        "close_approach_velocity_kilometers_per_second": null,
        "first_observation_date": "1950-06-19",
        "last_observation_date": "2025-02-18",
        "observations_used": 8019,
        "orbital_period": "816.3131500366128"
    }
]
